### DISCUSSION TITLE
Are hooks only for within the Pocketbase web admin?

### DISCUSSION THREAD

**Discussion by selfagency** [pocketbase/pocketbase#5153] [Q&A] [answered]
I added some basic code to a hooks file. When I startup Pocketbase, I see the `hello`. When I hard-refresh a page inside Pocketbase's web admin, I see the messages logged in the console. But when I use the API to query my collections from outside the web admin, I see nothing.

`pb_hooks/main.pb.js`

```
console.log('hello');

onCollectionsListRequest(e => {
  console.log('listreq', JSON.stringify(e.httpContext));
  console.log('listreq', JSON.stringify(e.collections));
  console.log('listreq', JSON.stringify(e.result));
});

onCollectionViewRequest(e => {
  console.log('viewreq', JSON.stringify(e.httpContext));
  console.log('viewreq', JSON.stringify(e.collections));
  console.log('viewreq', JSON.stringify(e.result));
});
```

**Comment by ganigeorgiev** [comment, marked as answer]
The Admin UI is a static client-side application that uses the same [Web APIs](https://pocketbase.io/docs/api-records/) under the hood.

The above hook fire only when interacting with the Admin UI because they are for the collection models, and collection models can be accessed or modified only by admins.

If you want to listen to the records events, please check https://pocketbase.io/docs/js-event-hooks/#record-crud-api-hooks.

**Reply by selfagency** [reply]
thanks again
